### PR TITLE
hit selection module

### PR DESCRIPTION
**Pull request by LaurenceKuhl**
This PR aims to add a new feature : hit selection. The aim would be to run an algorithm on found gene essentials and determine a p-value cutoff based on the degree of protein protein interaction network.

nf-core/crisprseq runs MAGeCK RRA, MAGeCK MLE, BAGEL2 or DRUGZ so I've plugged HITSELECTION after each of these processes and added a parameter giving the user the option to run hitselection or not.

It bases itself on an R script in the module hitselection.nf and on 2 assets : hgnc_complete_set and biogrid_hgncid

### CODE REVIEW

File: docs/output/screening.md [hunk 1]
```diff
@@ -190,6 +190,12 @@ For further reading and documentation see the [cutadapt helper page](https://cut
   - `*.txt`: Pathway view for top enriched pathways.
   - `*.png`: Pathway view for top enriched pathways.
 
+### HitSelection
+
+- `HitSelection`
+  - `*.png` : -logP value vs gene rank plot to determine the rank threshols
```
Comment (LaurenceKuhl) [inline docs/output/screening.md:196]:
```suggestion
  - `*.png` : -logP value vs gene rank plot to determine the rank thresholds
```

File: docs/usage/screening.md [hunk 2]
```diff
@@ -134,6 +134,18 @@ The contrast from reference to treatment should be ; separated
...
+
+Hitselection is a script for identifying rank thresholds for CRISPR screen results based on using the connectivity of subgraphs of protein-protein interaction (PPI) networks. The script is based on R and is also an implementation of RNAiCut (Kaplow et al., 2009), a method for estimating thresholds in RNAi data. The principle behind Hitselection is that true positive hits are densely connected in the PPI networks. The script runs a simulation based on Poisson distribution of the ranked screen gene list to calculate the -logP value for comparing the interconnectivity of the real subnetwork and the degree match random subnetwork of each gene, one by one. The degree of the nodes is used as the interconnectivity metric.
+
+To run Hitselection, you can specify '--hitselection' and it will automatically run on the gene essentiality algorithms you have chosen. The outputs are a png file containing the -logP value vs gene rank plot and a txt file containing all the -logP values, edge and average edge values and ranked gene symbols.
+
+## :warning: The hitselection algorithm is for the moment developped only for KO screens and requires the library to map to genes with an EntrezID.
```
Comment (LaurenceKuhl) [inline docs/usage/screening.md:145]:
```suggestion
## :warning: The hitselection algorithm is for the moment developped only for KO screens and requires the library to map to genes with an Homosapiens EntrezID.
```

**Review by WackerO — APPROVED** [review]
Left some small remarks, looks otherwise good so I'll directly approve :)

File: README.md [hunk 3]
```diff
@@ -67,10 +67,12 @@ For crispr screening:
...
 3. Optional: CNV correction and normalization with ([`CRISPRcleanR`](https://github.com/francescojm/CRISPRcleanR))
 4. Rank sgRNAs and genes ;
-   a. ([MAGeCK test](https://sourceforge.net/p/mageck/wiki/usage/#test))
-   b. ([MAGeCK mle](https://sourceforge.net/p/mageck/wiki/Home/#mle))
-   c. ([BAGEL2](https://github.com/hart-lab/bagel))
-5. Visualize analysis
+   - ([MAGeCK test](https://sourceforge.net/p/mageck/wiki/usage/#test))
+   - ([MAGeCK mle](https://sourceforge.net/p/mageck/wiki/Home/#mle))
+   - ([BAGEL2](https://github.com/hart-lab/bagel))
+   - ([DrugZ](https://github.com/hart-lab/drugz))
```
Comment (mirpedrol) [inline README.md:73]: Should this go under point 5?

File: docs/usage/screening.md [hunk 4]
```diff
@@ -134,6 +134,18 @@ The contrast from reference to treatment should be ; separated
...
+
+To run Hitselection, you can specify '--hitselection' and it will automatically run on the gene essentiality algorithms you have chosen. The outputs are a png file containing the -logP value vs gene rank plot and a txt file containing all the -logP values, edge and average edge values and ranked gene symbols.
+
+## :warning: The hitselection algorithm is for the moment developed only for KO screens and requires the library to map to genes with an Homosapiens EntrezID.
+
+## :warning: Please be advised that the Hitselection algorithm is time intensive and will make the pipeline run longer
```
Comment (mirpedrol) [inline docs/usage/screening.md:147]: To show a formatted warning box on the webside the notation is the following:
```
> [!WARNING]
> The hitselection algorithm is for the moment developed only for KO screens and requires the library to map to genes with an Homosapiens EntrezID.

> [!WARNING]
Please be advised that the Hitselection algorithm is time intensive and will make the pipeline run longer
```

**Review by mirpedrol — COMMENTED** [review]
Hi! I know this PR is already merged but I am leaving a couple of comments which I think should be fixed in the docs 🙂

File: README.md (hunk 3 above)
Comment (LaurenceKuhl) [reply README.md:73]: Actually no it's also a gene essentiality analysis but only on drug treated cell lines. It does something similar to MAGeCK MLE

**Pull request merged**